### ISSUE TITLE
Make map of if -> SvgElem public (or protected at least?)

### ISSUE THREAD

**Issue by shaurya947** [BigBadaboom/androidsvg#133]
I'm working on an app where I not only have to display a complex svg (for which this library is great!), but also parse the svg elements to do touch detection on individual shapes later..

I looked through the source files and it seems like this library is already internally doing what I would need to do for the parsing. Does it not make sense to somehow expose `idToElementMap` from `SVG` so that this use case could also be met. If not through a getter, then perhaps make it protected so that someone could extend SVG to gain access to it?

Happy to submit a PR!

**Comment by BigBadaboom** [comment]
If we did that, it would make most sense to expose `SVG.getElementById()` I think.  `idToElementMap` only contains those elements that have already been looked up with this function.  I presume you would also need me to make all of the element classes like `SVG.Path` public as well?

How were you planning to do the touch-to-element detection?  And why would you need this map for that?

I wasn't planning to get this in the 1.3 release (it's been delayed enough already :), But I suppose I could consider adding it.  How I was planning to implement this was to return the `id` attribute of the touched element (if it has one).

**Comment by BigBadaboom** [comment]
See also: issue #95

**Comment by shaurya947** [comment]
You're right -- `idToElementMap` doesn't get populated unless you do something like `renderViewToCanvas`. What I ended up doing was exposing `rootElement`, and all necessary internal classes like `Svg`, `Group`, `Rect`, `Polygon.` I believe exposing `getElementById` and other internal classes would have the same end result.

I am dealing with an svg file that has multiple layers (groups), and each group is made up of rects or polygons. When a user taps at a specific point, I need to find out the enclosing rect/polygon from each layer. Gaining access to `rootElement` helped me gain access to all the layers, and furthermore to the children in each layer. Then, for each rect/polygon, I created a corresponding Java `Polygon` object using https://github.com/mondain/android-components/blob/master/src/org/gregoire/android/graphics/Polygon.java

Finally, when the user taps, I call `Polygon.contains()` with the x and y of the tap for all the polygons in each layer until I find one from each layer.

**Comment by ShaMan123** [comment]
+1

**Comment by ShaMan123** [comment]
How about using the layer idea this way?
1. splitting the `SVG` element into bits
1. rendering each bit as an independent `SVGImageView`
1. reassembling in a `ViewGroup`

I guess this will allow to intercept touches on each part and will be responsive for transforms as any other view.

Will this work?
